### PR TITLE
added: play next context menu and action

### PR DESCRIPTION
**Pull request by tadly**
## Description

Based on: #14887

Adds a "Play next" context menu item for both video and music and a `PlayNext` action.

## Motivation and Context

I often see myself wanting to sneak in a track/video :)

## How Has This Been Tested?

Tested both the action (mapped to `ctrl+q`) and context menu for:
  * [x] Local video
  * [x] Local music
  * [x] Video addon (youtube)
  * [x] Music addon (gmusic)

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -2260,6 +2260,12 @@ msgstr ""
 
 #empty string with id 490
 
+#: xbmc/music/windows/GUIWindowMusicBase.cpp
+#: xbmc/video/windows/GUIWindowVideoBase.cpp
+msgctxt "#10008"
+msgid "Play next"
+msgstr ""
+
```
Comment (Rechi) [inline addons/resource.language.en_gb/resources/strings.po]: This file is sorted by ids, so move this to the correct place.

File: xbmc/dialogs/GUIDialogContextMenu.h [hunk 2]
```diff
@@ -92,6 +92,7 @@ enum CONTEXT_BUTTON { CONTEXT_BUTTON_CANCELLED = 0,
                       CONTEXT_BUTTON_EDIT_SORTTITLE,
                       CONTEXT_BUTTON_DELETE_ALL,
                       CONTEXT_BUTTON_HELP,
+                      CONTEXT_BUTTON_PLAY_NEXT
```
Comment (Rechi) [inline xbmc/dialogs/GUIDialogContextMenu.h]: For cleaner diffs
```suggestion
                      CONTEXT_BUTTON_PLAY_NEXT,
```

File: xbmc/input/actions/ActionIDs.h [hunk 3]
```diff
@@ -53,6 +53,7 @@
 #define ACTION_TOGGLE_SOURCE_DEST     32 //!< used to toggle between source view and destination view. Can be used in myfiles.xml window id=3
 #define ACTION_SHOW_PLAYLIST          33 //!< used to toggle between current view and playlist view. Can b used in all mymusic xml files
 #define ACTION_QUEUE_ITEM             34 //!< used to queue a item to the playlist. Can b used in all mymusic xml files
+#define ACTION_PLAY_NEXT_ITEM        259 //!< used to queue an item to the next position in the playlist
```
Comment (Rechi) [inline xbmc/input/actions/ActionIDs.h]: This is sorted by the number, so move to the correct place. Otherwise one may think it's unused later.

Comment (tadly) [reply xbmc/input/actions/ActionIDs.h]: It's kinda not sorted (see line 47 which is why I wasn't sure about it) but I'll move it anyways :)
Also gonna give it `247` as I noticed that one being free as well

Comment (tadly) [reply xbmc/input/actions/ActionIDs.h]: lol wow.
247 is exactly the idea which is out of line and I re-used it.

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by da-anda** [comment]
is it just me, or does `ACTION_PLAY_NEXT_ITEM` give a very wrong impression from it's name to what it does? It wont "play the next item" but "play the item next". So what about `ACTION_PLAY_ITEM_NEXT`?

**Comment by tadly** [comment]
Yeah, I remember not being happy with the wording either.
Totally forgot about asking.

I'm fine with changing it to `ACTION_PLAY_ITEM_NEXT`

**Comment by tadly** [comment]
Be as nit-picky as you must, I don't mind :)

Would this imply to change everything from "Play next" to "Queue next"?
Because "Play next" within the UI does make more sense IMO.

**Comment by DaveTBlake** [comment]
Yes in UI "Play Next" makes most sense. My comment only applies internally to code and consistency with how we have named other actions and buttons. But not a big deal, @da-anda comment just  pushed me into mentioning it.

**Comment by tadly** [comment]
commit hash mismatch
jenkins build this please

**Comment by tadly** [comment]
Btw. just curious.
This being a feature while we being in RC by now means it's not gonna make it into the final v18 release right?

**Comment by da-anda** [comment]
> Btw. just curious.
> This being a feature while we being in RC by now means it's not gonna make it into the final v18 release right?

depends on how much beer you send in @MartijnKaijser 's direction :)

**Comment by tadly** [comment]
I don't wanna sture up anything.
It's my fault for coming up with this so late (should not have forgotten about it :))

**Comment by MartijnKaijser** [comment]
If it get approval from another dev it can be included

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 4]
```diff
@@ -871,6 +879,8 @@ void CGUIWindowVideoBase::GetContextButtons(int itemNumber, CContextButtons &but
           buttons.Add(CONTEXT_BUTTON_PLAY_AND_QUEUE, 13412);
         else
           buttons.Add(CONTEXT_BUTTON_PLAY_ONLY_THIS, 13434);
+
+        buttons.Add(CONTEXT_BUTTON_PLAY_NEXT, 10008);      // Play next
```
Comment (Rechi) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: This isn't the correct place, see the description of the if.
https://github.com/xbmc/xbmc/blob/ed0a72307046cd17c33ad8f548a6de8276150e15/xbmc/video/windows/GUIWindowVideoBase.cpp#L865-L867

Comment (tadly) [reply xbmc/video/windows/GUIWindowVideoBase.cpp]: Fair point.
Does moving it [here](https://github.com/xbmc/xbmc/blob/0f3197d73d32b8bff21388339350ee7e21f6fb8c/xbmc/video/windows/GUIWindowVideoBase.cpp#L847) sound good?

Comment (Rechi) [reply xbmc/video/windows/GUIWindowVideoBase.cpp]: Yes, that's the place I would have expected it.

**Comment by tadly** [comment]
commit hash mismatch
jenkins build this please

**Comment by Rechi** [comment]
@tadly nothing changed in your last PR updated, was this by intention?

**Comment by tadly** [comment]
No... No it was not ._. Forgot to `add` before `commit --amend` :/

**Review by Rechi — APPROVED** [review]

**Comment by tadly** [comment]
Thanks a bunch to all involved.
I **really** appreciate all the help and this even getting merged for v18 :)

**Pull request merged**